sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -189,14 +189,18 @@ case class DropTableCommand(
     if (!catalog.isTemporaryTable(tableName) && catalog.tableExists(tableName)) {
       // If the command DROP VIEW is to drop a table or DROP TABLE is to drop a view
       // issue an exception.
-      catalog.getTableMetadata(tableName).tableType match {
-        case CatalogTableType.VIEW if !isView =>
-          throw new AnalysisException(
-            "Cannot drop a view with DROP TABLE. Please use DROP VIEW instead")
-        case o if o != CatalogTableType.VIEW && isView =>
-          throw new AnalysisException(
-            s"Cannot drop a table with DROP VIEW. Please use DROP TABLE instead")
-        case _ =>
+      try {
+        catalog.getTableMetadata(tableName).tableType match {
+          case CatalogTableType.VIEW if !isView =>
+            throw new AnalysisException(
+              "Cannot drop a view with DROP TABLE. Please use DROP VIEW instead")
+          case o if o != CatalogTableType.VIEW && isView =>
+            throw new AnalysisException(
+              s"Cannot drop a table with DROP VIEW. Please use DROP TABLE instead")
+          case _ =>
+        }
+      } catch {
+          case e: QueryExecutionException => log.warn(e.toString, e)
```

Comment:
@gatorsmile  and @Davies  ; In that case, any suggestions to mock the metadata corrupt scenario?